### PR TITLE
Comparators + Tests

### PR DESCRIPTION
**Pull request by jacobrs**
Piece comparator and Board comparator accompanied by tests.

### CODE REVIEW

File: tests/datastructures/BoardTests.cpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+TEST(BoardTests, BoardComparatorDifferentBoards) {
+
+    auto board1 = new Board();
+    auto board2 = new Board();
+    Position from = {'E', 2};
+    Position to = {'E', 4};
+    board2->movePiece(from, to, WHITE);
+
+    ASSERT_FALSE(*board1 == *board2);
+    ASSERT_TRUE(*board1 != *board2);
```
Comment (benrs) [inline tests/datastructures/BoardTests.cpp:23]: What's the point of two asserts here?

**Comment by benrs** [comment]
Looks good but why are you double asserting everything?

File: tests/datastructures/BoardTests.cpp (hunk 1 above)
Comment (jacobrs) [reply tests/datastructures/BoardTests.cpp:23]: since I overload both operators I need to test both operators.

Comment (benrs) [reply tests/datastructures/BoardTests.cpp:23]: Touche, I thought it would implicitly use !(==)

**Review by benrs — APPROVED** [review]

**Pull request merged**